pynest/examples/pynest-microcircuit/network.py

```diff
@@ -0,0 +1,347 @@
+        print 'Poisson background input connected'
+
+    # the thalamic neurons are connected to the different populations
+    def connect_thalamus(self):
+        for i, target_pop in enumerate(self.pops):
+                conn_dict_th = {'rule': 'fixed_total_number',
+                                'N': int(self.nr_synapses_th[i])}
+                syn_dict_th = {'weight': {'distribution': 'normal_clipped',
+                                          'mu': self.thalamic_weight,
+                                          'sigma': (self.thalamic_weight *
+                                                    self.net_dict['PSP_sd']),
+                                          'low': 0.0},
+                               'delay': {'distribution': 'normal_clipped',
+                                         'mu': self.stim_dict['delay_th'][i],
+                                         'sigma': stim_dict['delay_th_sd'][i],
+                                         'low': self.sim_resolution}}
+                nest.Connect(self.thalamic_population, target_pop,
+                             conn_spec=conn_dict_th, syn_spec=syn_dict_th)
+        print 'Thalamus connected'
+
+    # the dc generators are connected to the neuronal populations
+    def connect_dc_generator(self):
+        for i, target_pop in enumerate(self.pops):
+            if self.stim_dict['dc_input']:
+                nest.Connect(self.dc[i], target_pop)
+        print 'DC Generator connected'
+
+    # the recording devices are connected to the neuronal populations
+    def connect_devices(self):
+        for i, target_pop in enumerate(self.pops):
+            if 'voltmeter' in self.net_dict['rec_dev']:
+                nest.Connect(self.voltmeter[i], target_pop)
+            if 'spike_detector' in self.net_dict['rec_dev']:
+                nest.Connect(target_pop, self.spikedetector[i])
+        print '%s of 2 Devices connected' % (len(self.net_dict['rec_dev']))
+
+    def connect(self):
+        """ Execute subfunctions of the network.
+        Create populations, devices and inputs, connect the nodes, etc.
+        """
+        self.setup()
+        self.create_populations()
+        self.create_devices()
+        self.create_thalamic_input()
+        self.create_poisson()
+        self.create_dc_generator()
+        self.create_connections()
+        if self.net_dict['poisson_input']:
+            self.connect_poisson()
+        if self.stim_dict['thalamic_input']:
+            self.connect_thalamus()
+        if self.stim_dict['dc_input']:
+            self.connect_dc_generator()
+        self.connect_devices()
+
+    # this function simulates the network
+    def simulate(self):
+        nest.Simulate(self.sim_dict['t_sim'])
+
+    def plot_spikes(self):
+        dSD = nest.GetStatus(self.spikedetector, keys='events')[0]
+        evs = dSD['senders']
+        ts = dSD['times']
+        plt.figure(1)
+        plt.plot(ts, evs, '.')
+        plt.show()
```

Comment:
In the current state, this function does not work:
- self.spikedetector is a list of tuples of GIDs. `nest.GetStatus` needs a list or a tuple of GIDs so you should adapt self.spikedetector accordingly.
- currently, the spike detector to not record to memory so that the status dictionaries of the spike detectors do not contain the spikes
- Please rewrite this plot function so that it returns a raster plot with properly stacked populations with different colors for E and I populations (like in the manuscript) and a bar plot with the population rates. You can take the PyNN microcircuit as a template.
- Finally, `example.py` should execute the plot function by default.